### ISSUE TITLE
Incompatible with Terrestria and Traverse

### ISSUE THREAD

**Issue by forkiesassds** [caeruleusDraconis/world-preview#8]
With either one of these mods (since they use Biolith) the preview to not work at all
```
[13:44:11] [Render thread/INFO]: [STDERR]: java.util.NoSuchElementException: No value present
[13:44:11] [Render thread/INFO]: [STDERR]: 	at java.base/java.util.Optional.orElseThrow(Optional.java:377)
[13:44:11] [Render thread/INFO]: [STDERR]: 	at com.terraformersmc.biolith.impl.biome.BiomeCoordinator.getBiomeLookupOrThrow(BiomeCoordinator.java:50)
[13:44:11] [Render thread/INFO]: [STDERR]: 	at com.terraformersmc.biolith.impl.biome.OverworldBiomePlacement.writeBiomeEntries(OverworldBiomePlacement.java:88)
[13:44:11] [Render thread/INFO]: [STDERR]: 	at net.minecraft.class_4766.wrapOperation$bpo000$biolith$injectEntries(class_4766.java:556)
[13:44:11] [Render thread/INFO]: [STDERR]: 	at net.minecraft.class_4766.method_49506(class_4766.java:50)
[13:44:11] [Render thread/INFO]: [STDERR]: 	at net.minecraft.class_4766.method_49494(class_4766.java:58)
[13:44:11] [Render thread/INFO]: [STDERR]: 	at net.minecraft.class_1966.method_49495(class_1966.java:31)
[13:44:11] [Render thread/INFO]: [STDERR]: 	at com.google.common.base.Suppliers$NonSerializableMemoizingSupplier.get(Suppliers.java:183)
[13:44:11] [Render thread/INFO]: [STDERR]: 	at net.minecraft.class_1966.method_28443(class_1966.java:41)
[13:44:11] [Render thread/INFO]: [STDERR]: 	at net.minecraft.class_7869.method_46706(class_7869.java:79)
[13:44:11] [Render thread/INFO]: [STDERR]: 	at net.minecraft.class_7869.method_46699(class_7869.java:66)
[13:44:11] [Render thread/INFO]: [STDERR]: 	at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:178)
[13:44:11] [Render thread/INFO]: [STDERR]: 	at java.base/java.util.AbstractList$RandomAccessSpliterator.forEachRemaining(AbstractList.java:722)
[13:44:11] [Render thread/INFO]: [STDERR]: 	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509)
[13:44:11] [Render thread/INFO]: [STDERR]: 	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
[13:44:11] [Render thread/INFO]: [STDERR]: 	at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:921)
[13:44:11] [Render thread/INFO]: [STDERR]: 	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
[13:44:11] [Render thread/INFO]: [STDERR]: 	at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:682)
[13:44:11] [Render thread/INFO]: [STDERR]: 	at net.minecraft.class_7869.method_46702(class_7869.java:67)
[13:44:11] [Render thread/INFO]: [STDERR]: 	at net.minecraft.class_2794.method_46696(class_2794.java:108)
[13:44:11] [Render thread/INFO]: [STDERR]: 	at caeruleusTait.world.preview.backend.worker.SampleUtils.<init>(SampleUtils.java:176)
[13:44:11] [Render thread/INFO]: [STDERR]: 	at caeruleusTait.world.preview.backend.WorkManager.changeWorldGenState(WorkManager.java:126)
[13:44:11] [Render thread/INFO]: [STDERR]: 	at caeruleusTait.world.preview.client.gui.screens.PreviewContainer.updateSettings_real(PreviewContainer.java:431)
[13:44:11] [Render thread/INFO]: [STDERR]: 	at caeruleusTait.world.preview.client.gui.screens.PreviewContainer.lambda$updateSettings$21(PreviewContainer.java:351)
[13:44:11] [Render thread/INFO]: [STDERR]: 	at java.base/java.util.concurrent.CompletableFuture$UniAccept.tryFire(CompletableFuture.java:718)
[13:44:11] [Render thread/INFO]: [STDERR]: 	at java.base/java.util.concurrent.CompletableFuture$Completion.run(CompletableFuture.java:482)
[13:44:11] [Render thread/INFO]: [STDERR]: 	at net.minecraft.class_1255.method_18859(class_1255.java:156)
[13:44:11] [Render thread/INFO]: [STDERR]: 	at net.minecraft.class_4093.method_18859(class_4093.java:23)
[13:44:11] [Render thread/INFO]: [STDERR]: 	at net.minecraft.class_1255.method_16075(class_1255.java:130)
[13:44:11] [Render thread/INFO]: [STDERR]: 	at net.minecraft.class_1255.method_5383(class_1255.java:115)
[13:44:11] [Render thread/INFO]: [STDERR]: 	at net.minecraft.class_310.method_1523(class_310.java:1175)
[13:44:11] [Render thread/INFO]: [STDERR]: 	at net.minecraft.class_310.method_1514(class_310.java:802)
[13:44:11] [Render thread/INFO]: [STDERR]: 	at net.minecraft.client.main.Main.main(Main.java:250)
[13:44:11] [Render thread/INFO]: [STDERR]: 	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
[13:44:11] [Render thread/INFO]: [STDERR]: 	at java.base/java.lang.reflect.Method.invoke(Method.java:578)
[13:44:11] [Render thread/INFO]: [STDERR]: 	at org.quiltmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:524)
[13:44:11] [Render thread/INFO]: [STDERR]: 	at org.quiltmc.loader.impl.launch.knot.Knot.launch(Knot.java:82)
[13:44:11] [Render thread/INFO]: [STDERR]: 	at org.quiltmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:28)
[13:44:11] [Render thread/INFO]: [STDERR]: 	at org.prismlauncher.launcher.impl.StandardLauncher.launch(StandardLauncher.java:87)
[13:44:11] [Render thread/INFO]: [STDERR]: 	at org.prismlauncher.EntryPoint.listen(EntryPoint.java:125)
[13:44:11] [Render thread/INFO]: [STDERR]: 	at org.prismlauncher.EntryPoint.main(EntryPoint.java:70)
```

**Comment by caeruleusDraconis** [comment]
Which Minecraft version are you using? Fabric / Forge?

**Comment by forkiesassds** [comment]
1.20.1, on Quilt using QFAPI (targeting FAPI 0.88.1)

**Comment by forkiesassds** [comment]
Seems to happen only in the world creation screen, I guess whatever initialization it has to do when generating the world doesn't happen in the world creation screen.

**Comment by caeruleusDraconis** [comment]
This is likely because a fake Minecraft world is "created" in the world creation screen, which apparently does not trigger the biolith initialization...

I will see if I can work around this.

**Comment by caeruleusDraconis** [comment]
Should be fixed with the 1.1.1 release: https://modrinth.com/mod/world-preview/version/1.1.1

Feel free to reopen if the issue still persists.
